src/dictionaries.cpp

```diff
@@ -18,17 +18,31 @@ uint32_t BasicDictionary::code(const std::string& name) const
 
 void BasicDictionary::add(uint32_t code, const std::string& name)
 {
-    m_rightDict.emplace(code, name);
+    for (const auto& entry: m_rightDict)
+        if (entry.second == name && entry.first != code)
+            throw RadProto::Exception(RadProto::Error::suchAttributeNameAlreadyExists, "Function add. Attribute name " + name + " already exists with code " + std::to_string(entry.first));
```

Comment:
Instead of 'Function add' I would use something like '[BasicDictionary::add]'. First, it is more specific. Second, it clearly separates location information from the message itself.

Same below.